### PR TITLE
Tell dh_shlibdeps where to find rviz_ogre_vendor plugin libraries

### PR DESCRIPTION
**Pull request by hidmic**
As of https://github.com/ros2/rviz/pull/688, `rviz_ogre_vendor` plugin libraries that depend on other plugin libraries no longer have an RPATH for `dpkg-shlibdeps` to find them.

### CODE REVIEW

File: debian/rules.em
Comment (clalancette) [inline debian/rules.em:54]: So this change makes it so that dh_shlibdeps can find the ogre vendor, and hence finish building.  This is likely a dumb question, but I'll ask it anyway: can rviz2 find Ogre at runtime?  I just want to make sure rviz2 wasn't depending on the RPATH to find it.

Comment (hidmic) [reply debian/rules.em:54]: I didn't run into trouble when testing https://github.com/ros2/rviz/pull/688. It couldn't depend on those RPATHs though, those are absolute paths in the build host filesystem. In the example I posted on that PR:

``` 
$ readelf -d /opt/ros/rolling/opt/rviz_ogre_vendor/lib/OGRE/Plugin_OctreeZone.so

Dynamic section at offset 0xdd70 contains 31 entries:
  Tag        Type                         Name/Value
 0x0000000000000001 (NEEDED)             Shared library: [Plugin_PCZSceneManager.so.1.12.1]
 0x0000000000000001 (NEEDED)             Shared library: [libOgreMain.so.1.12.1]
 0x0000000000000001 (NEEDED)             Shared library: [libstdc++.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libm.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libgcc_s.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
 0x000000000000000e (SONAME)             Library soname: [Plugin_OctreeZone.so.1.12.1]
 0x000000000000001d (RUNPATH)            Library runpath: [/tmp/binarydeb/ros-rolling-rviz-ogre-vendor-8.5.0/obj-x86_64-linux-gnu/ogre_install/lib:/tmp/binarydeb/ros-rolling-rviz-ogre-vendor-8.5.0/obj-x86_64-linux-gnu/ogre_install/lib/OGRE]
...
```
you can see `/tmp/binarydeb/ros-rolling-rviz-ogre-vendor-8.5.0/obj-x86_64-linux-gnu/` paths, ignored everywhere but when running `dpkg-shlibdeps`.

Comment (nuclearsandwich) [reply debian/rules.em:54]: > can rviz2 find Ogre at runtime? I just want to make sure rviz2 wasn't depending on the RPATH to find it.

The environment hook supplied by the vendor package provides the library location for all other use cases. https://github.com/ros2/rviz/tree/ros2/rviz_ogre_vendor/env_hook

Comment (nuclearsandwich) [reply debian/rules.em:54]: It looks like these plugins are vendored within the vendored ogre package. Since the environment hook doesn't add this subdirectory presumably these plugins are only ever linked by ogre itself and it has configuration to know where to find them.

Otherwise how would e.g. rviz know to find libraries in `/opt/ros/rolling/lib/rviz_ogre_vendor/lib/OGRE` vs /opt/ros/rolling/lib/rviz_ogre_vendor/lib`?

Comment (clalancette) [reply debian/rules.em:54]: > The environment hook supplied by the vendor package provides the library location for all other use cases. https://github.com/ros2/rviz/tree/ros2/rviz_ogre_vendor/env_hook

Ah, cool.  Thanks for that link, that makes sense.

One other dumb question: why is the env_hook setting the `LD_LIBRARY_PATH` to only `PREFIX/opt/rviz_ogre_vendor/lib`, while this change is additionally adding `PREFIX/opt/rviz_ogre_vendor/lib/OGRE`?

Comment (hidmic) [reply debian/rules.em:54]: That hook doesn't point to the plugin libraries. I haven't found what pulls those in.

Comment (clalancette) [reply debian/rules.em:54]: All right, that's fair.  Thanks for the feedback.  I have no objection to the patch, so I'm going to approve.  It would be good to get another approval from @nuclearsandwich before merging.

**Review by clalancette — APPROVED** [review]

**Review by nuclearsandwich — APPROVED** [review]
Approving what works empirically. If we learn more about whether the OGRE plugin lib directory should be linkable by the rest of the system we ought to revisit this patch.

**Comment by hidmic** [comment]
@nuclearsandwich binary job is still failing, but it doesn't look that this patch took place. 

From [Rbin_uF64__rviz_ogre_vendor__ubuntu_focal_amd64__binary/41](https://build.ros2.org/view/Rbin_uF64/job/Rbin_uF64__rviz_ogre_vendor__ubuntu_focal_amd64__binary/41/console): 

```sh
dh_shlibdeps -l/tmp/binarydeb/ros-rolling-rviz-ogre-vendor-8.6.0/debian/ros-rolling-rviz-ogre-vendor//opt/ros/rolling/lib/:/tmp/binarydeb/ros-rolling-rviz-ogre-vendor-8.6.0/debian/ros-rolling-rviz-ogre-vendor//opt/ros/rolling/opt/rviz_ogre_vendor/lib/
```

Did I merge this patch into the wrong branch?

**Comment by clalancette** [comment]
I *think* we need to do a re-release of the package in order to get this in.  Particularly because this is patching the empy file, not the one that is actually used.

**Comment by nuclearsandwich** [comment]
> I _think_ we need to do a re-release of the package in order to get this in

This is correct and was also the conclusion of an offline discussion.

> Particularly because this is patching the empy file, not the one that is actually used.

This wouldn't make a difference because the build farm jobs do not fetch from branches, but rather fixed tags based on the release version in distribution.yaml so in order for any changes from the release repository branches to be propagated a release needs to be performed. Source changes and version bumps are not required, bloom will automatically create a package revision (also called a debinc in some parts of the bloom source) when re-releasing the latest version of an already released repository and changes to the packaging setup are exactly what package revisions are for so doing so is not a cheat or elision of the rules. It's just a rarer than normal task since most changes to ROS packages happen in the source and not in packaging.

**Comment by hidmic** [comment]
> This is correct and was also the conclusion of an offline discussion.

Indeed ! Sorry for the noise. Builds are healthy now https://build.ros2.org/view/Rbin_uF64/job/Rbin_uF64__rviz_ogre_vendor__ubuntu_focal_amd64__binary/42/ :)

**Pull request merged**
